src/carousel/carousel.ts

```diff
@@ -274,7 +274,15 @@ export class NgbCarousel implements AfterContentChecked,
       });
     }
 
-    this.slides.changes.pipe(takeUntil(this._destroy$)).subscribe(() => this._cd.markForCheck());
+    this.slides.changes.pipe(takeUntil(this._destroy$)).subscribe(() => {
+      this._cd.detectChanges();
+      if (this.activeId) {
```

Comment:
What's with `'0'` id ?
Also do we care about this `if (this.activeId)` at all ?